go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/kind
 
-go 1.14
+go 1.20
```

Comment:
Any specific reason that was still in 1.14?

Reply:
Yes
/hold
See: https://github.com/kubernetes-sigs/kind/pull/3157

Note that go.mod go version implies the go feature level, NOT the go version used for building. 

That would be .go-version here in most cases.

Reply:
@BenTheElder i will drop this change for now to not block this PR, is that ok?

Reply:
Yes, please revert changes to the go.mod, we're probably going to bump to 1.16 soon but we're not using any go 1.20 features and this is risky for library consumers